cmake/templates/pkgConfig.cmake.in

```diff
@@ -116,18 +137,31 @@ foreach(library ${libraries})
     list(APPEND @PROJECT_NAME@_LIBRARIES ${library})
   elseif(TARGET ${library})
     list(APPEND @PROJECT_NAME@_LIBRARIES ${library})
+  elseif(IS_ABSOLUTE ${library} AND CMAKE_CROSSCOMPILING)
+    get_filename_component(lib_path ${library} DIRECTORY)
+    get_filename_component(lib_name ${library} NAME)
+    set(lib "lib-NOTFOUND")
+    # use find_library to locate library given by absolute path nested under potential root path
+    find_library(lib ${lib_name}
+      PATHS ${lib_path}
+      NO_DEFAULT_PATH ${_find_root_path_both})
+    if(NOT lib)
+      message(FATAL_ERROR "Project '${PROJECT_NAME}' tried to find library '${library}'.  The library is neither a target nor built/installed properly.  Did you compile project '@PROJECT_NAME@'?  Did you find_package() it before the subdirectory containing its code is included?")
+    endif()
+    _list_append_unique(@PROJECT_NAME@_LIBRARY_DIRS ${lib_path})
+    list(APPEND @PROJECT_NAME@_LIBRARIES ${lib})
   elseif(IS_ABSOLUTE ${library})
     list(APPEND @PROJECT_NAME@_LIBRARIES ${library})
   else()
     set(lib_path "")
-    set(lib "${library}-NOTFOUND")
+    set(lib "lib-NOTFOUND")
     # since the path where the library is found is returned we have to iterate over the paths manually
     foreach(path @PKG_CONFIG_LIB_PATHS@)
       find_library(lib ${library}
         PATHS ${path}
-        NO_DEFAULT_PATH NO_CMAKE_FIND_ROOT_PATH)
+        NO_DEFAULT_PATH ${_find_root_path_both})
```

Comment:
Changing the argument in this PR does change the logic where the library might be found. Right? As far as I understand the code the argument is being passed so that the library is only found in the locations specified by `PKG_CONFIG_LIB_PATHS`. So removing the argument would change that in non cross-compile scenarios. Why should that be done? What is the intention of changing this line? Can you describe a case where the current behavior is wrong?